### DISCUSSION TITLE
Small bug in Quick disconnect (on Bluetooth)

### DISCUSSION THREAD

**Discussion by anwar-alsilwy** [nefarius/DsHidMini#187] [Support]
First of all, thank you for this beautiful project.
I have fake ps3 controller (PANHAI) and it's working great without issues.
But when I Quick disconnect (on Bluetooth) by pressing L1 + R1 + PS, it disconnect and reconnect immediately,
I know that I can make it disconnect completely from DSHMC, but I also need L1 + R1 + PS, it's so beautiful feature.
Cheers.

![20220206_184252](https://user-images.githubusercontent.com/71539261/152688857-14fede5e-ee1f-41f3-b7a3-0c37cd4ad1ec.jpg)

**Comment by nefarius** [comment]
If it reconnects again, it's the controllers doing, so not much the driver can do against I'm afraid.

**Reply by anwar-alsilwy** [reply]
I have 2 original PS3 controllers without this issue, and 4 new PANHAI controllers which all of them keeps reconnect immediately after I quick disconnect them by L1 + R1 + PS, but they only disconnect completely from DSHMC UI (disconnect option).

**Reply by nefarius** [reply]
Yeah so as I said, controller issue.

**Reply by Kanuan** [reply]
Since it works as normal if you disconnect through the DSHMC UI then it seems like the controller is detecting "PS" is being pressed and turning itself on again as soon as the controller is disconnected. If this is the case then it could only be fixed by software if the auto-disconnect combo were to be changed.

**Reply by nefarius** [reply]
Exactly, the controller is probably designed so badly it just keeps waking up again even if you don't press, release and press PS again but hold it down.

**Reply by anwar-alsilwy** [reply]
So, What the solution you prefer??
Can the developers change disconnect combo L1 + R1 + PS to anything else, to prevent controllers from reconnect, in example ( L1 + R1 + Select).

**Reply by mehmetcb02** [reply]
I had this exact problem at my PANHAI controller too. What i found as a workaround is if you manage to stop pressing buttons right after the moment the controller disconnects it'll not reconnect. But you have to be very precise and try to memorize how many seconds it takes to disconnect after the moment you start pressing the combo buttons. And of course YMMV.

**Reply by anwar-alsilwy** [reply]
I tried that many times, but without luck,
Do you think if the developer change the quick-disconnect combo to (e.g L1 + R1 + Select) will solve the issue ???

**Reply by nefarius** [reply]
Would probably fix it I'd assume.

**Reply by mehmetcb02** [reply]
It will probably solve but i don't think Nef will bother himself to change it since it's a knockoff controller and support for them is kinda limited? I've never experienced this on any Sony Genuine controller and i was able to escape this situation with the workaround i mentioned previously on my PANHAI controller so your best bet is keep trying it i guess.

**Reply by anwar-alsilwy** [reply]
Ok, fine
I hope you Implement the solution in next release
Thanks and I'm sorry if I wasted your time.

**Reply by nefarius** [reply]
It's not a huge change but users accustom to the year long existing combo certainly would be thrown off if I just slip the change in, even if mentioned in the change log. Will need to think about what's best to do, maybe do both? Keep L1 + R1 + PS and make it also listen to L1 + R1 + Select?

**Reply by landozz** [reply]
> It's not a huge change but users accustom to the year long existing combo certainly would be thrown off if I just slip the change in, even if mentioned in the change log. Will need to think about what's best to do, maybe do both? Keep L1 + R1 + PS and make it also listen to L1 + R1 + Select?

Im having a different problem with the quick disconnect and I think making l1+r1+select another shortcut is a good solution

**Reply by anwar-alsilwy** [reply]
Yes, make them both is a good solution.

**Comment by tsankuanglee** [comment]
I'm sharing my successful experience of connecting two PANHAI controllers via bluetooth. This is from my (many many) trials and errors. Maybe somebody will find this helpful.

I followed the standard guides online that are intended for the authentic PS3 controller, but the PANHAI controller would just disconnect itself a few seconds after I unplugged the USB cable. The following is what I did to avoid that problem. (I only had to do the following once. After that, I can just press the PS button and the controller will immediately connect to Windows via Bluetooth.)

1. I only have `DsHidMini` installed (I don't use `Ds4Windows`, but since we are dealing with the initial Bluetooth connection here, I don't think `Ds4Windows` is relevant anyway). For the installation, I just followed the standard guide: unpack the released zip; install the two *ini files; reboot. (This was tested under Windows 10, but Windows 11 should work as well.)

2. Run `DSHMC.exe` (not sure whether it's necessary to run it with Administrator privilege; I didn't, but since I had tried so many times, maybe one of them did the trick? It's harmless to do it anyway.)

3. Connect ONE PANHAI controller via USB. It should show up in `DSHMC`'s window as a USB connected controller. All 4 LEDs will blink continuously _at the same time_.

4. Press the PS button on the controller (I long pressed it, but I don't think it matters). The 4 LEDs on the front will blink _one by one_, cyclically.

5. (**IMPORTANT**!) Wait until only 1 LED stays solid lit. _It took me a few minutes_. Be patient here.

6. Unplug the USB cable from this controller (let's call it Controller 1).

7. DsHidMini will then do its magic and connect the controller via bluetooth.

8. (**IMPORTANT**!) Wait until only 1 LED stays solid lit. It took me a few seconds. Be patient here.

9. Controller 1 is now successfully setup. Leave it connected (via Bluetooth). For controller 2, follow step 3-8 as well.

Both PANHAI controllers are now properly setup, and are connected to Windows via Bluetooth, ready to use.

When I am done using them, I just turned off my Bluetooth and the controllers would go to sleep.

Now that both controllers are properly setup, the steps above are no longer needed (well obviously `DsHidMini` needs to remain installed). To use them again, I just press their PS buttons, and voila!

I think this peculiar behavior is due to PANHAI's specific firmware design. Maybe we just need to wait until something times out. :shrug:  

Thanks again for this wonderful project! It really made the fun accessible!
